### DISCUSSION TITLE
Whether to support the rest client that supports unix sockets

### DISCUSSION THREAD

**Discussion by wzzc-dev** [quarkusio/quarkus#29488] [Q&A]
```
security-api/mp-rest/url: unix:///opt/tmp/eulixspace-security-agent.sock
```

I have added the --enable-url-protocols=unix parameter to the configuration file

but error：

`Caused by: java.net.MalformedURLException: Accessing an URL protocol that was not enabled. The URL protocol unix is not tested and might not work as expected. It can be enabled by adding the --enable-url-protocols=unix option to the native-image command`

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
@cescoffier does the Vert.x HTTP Client support the UNIX socket protocol?

**Reply by cescoffier** [reply]
Yes, but only in JVM mode and you need to enable native transport (and pick the one you want - epoll I presume).

See: https://quarkus.io/guides/vertx-reference#native-transport

**Reply by geoand** [reply]
🙏

**Reply by dorianm** [reply]
We use Quarkus in native mode, do you have a recommendation to call an HTTP service through a Unix Socket in this case? I was considering using Vert.x HTTP Client until I saw this discussion.

**Comment by wzzc-dev** [comment]
Native transports are not supported in GraalVM produced binaries.

**Reply by cescoffier** [reply]
Yes, at the moment, we do not support domain sockets in native. It requires making Netty native transport work in native executables, and we never prioritized that work. Native transport brings a concurrency benefit that gets a bit reduced by the native compilation.

**Reply by dmlloyd** [reply]
Since Java 17 is our baseline, could/should we look into Netty using NIO support for UNIX domain sockets (which is available since Java 16)?

**Reply by cescoffier** [reply]
@dmlloyd Yes, it's something I added to my to-do list almost a year ago...
